cf-agent/verify_files.c

```diff
@@ -559,6 +583,74 @@ static PromiseResult VerifyFilePromise(EvalContext *ctx, char *path, const Promi
 
 /*****************************************************************************/
 
+static PromiseResult WriteContentFromString(EvalContext *ctx, const char *path, const Attributes *attr,
+                                            const Promise *pp)
+{
+    assert(pp != NULL);
+    assert(attr != NULL);
+
+    PromiseResult result = PROMISE_RESULT_NOOP;
+
+    if (attr->content == NULL)
+    {
+        RecordFailure(ctx, pp, attr, "'content' not set for promiser: '%s'", pp->promiser);
+        return PromiseResultUpdate(result, PROMISE_RESULT_FAIL);
+    }
+
+    unsigned char existing_content_digest[EVP_MAX_MD_SIZE + 1] = { 0 };
+    if (access(pp->promiser, R_OK) == 0)
+    {
+        HashFile(pp->promiser, existing_content_digest, CF_DEFAULT_DIGEST,
+                 FileNewLineMode(path) == NewLineMode_Native);
```

Comment:
@vpodzime Is this the correct way to get the new line mode? Previously this was done inside EditContext.

Reply:
Yes, that should be correct. We don't want to replace the file contents of a file on windows over and over just because when writing the contents, windows add their newlines.